File: lib/backend/supabase/database/tables/identity_cards.dart
import '../database.dart';

class IdentityCardsTable extends SupabaseTable<IdentityCardsRow> {
  @override
  String get tableName => 'identity_cards';

  @override
  IdentityCardsRow createRow(Map<String, dynamic> data) =>
      IdentityCardsRow(data);
}

class IdentityCardsRow extends SupabaseDataRow {
  IdentityCardsRow(super.data);

  @override
  SupabaseTable get table => IdentityCardsTable();

  int get id => getField<int>('id')!;
  set id(int value) => setField<int>('id', value);

  DateTime get createdAt => getField<DateTime>('created_at')!;
  set createdAt(DateTime value) => setField<DateTime>('created_at', value);

  String get createdBy => getField<String>('created_by')!;
  set createdBy(String value) => setField<String>('created_by', value);

  DateTime? get updatedAt => getField<DateTime>('updated_at');
  set updatedAt(DateTime? value) => setField<DateTime>('updated_at', value);

  String? get updatedBy => getField<String>('updated_by');
  set updatedBy(String? value) => setField<String>('updated_by', value);

  String get name => getField<String>('name')!;
  set name(String value) => setField<String>('name', value);

  String? get note => getField<String>('note');
  set note(String? value) => setField<String>('note', value);

  String? get country => getField<String>('country');
  set country(String? value) => setField<String>('country', value);

  String? get expiryDate => getField<String>('expiry_date');
  set expiryDate(String? value) => setField<String>('expiry_date', value);

  String get identityCardNumber => getField<String>('identity_card_number')!;
  set identityCardNumber(String value) =>
      setField<String>('identity_card_number', value);

  String? get identityCardType => getField<String>('identity_card_type');
  set identityCardType(String? value) =>
      setField<String>('identity_card_type', value);

  String? get issueDate => getField<String>('issue_date');
  set issueDate(String? value) => setField<String>('issue_date', value);

  String get nameOnCard => getField<String>('name_on_card')!;
  set nameOnCard(String value) => setField<String>('name_on_card', value);

  String? get state => getField<String>('state');
  set state(String? value) => setField<String>('state', value);
}

File: rust/src/models/identity_cards.rs
use chrono::NaiveDateTime;
use crudlf_derive::{SqliteDelete, SqliteInsert, SqliteListFilter, SqliteSelect, SqliteUpdate};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(
    Debug,
    Clone,
    PartialEq,
    Serialize,
    Deserialize,
    FromRow,
    SqliteInsert,
    SqliteSelect,
    SqliteUpdate,
    SqliteDelete,
    SqliteListFilter,
)]
#[crudlf(table_name = "identity_cards")]
pub struct IdentityCard {
    #[serde(rename = "id")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub id: Option<String>,

    #[serde(rename = "created_at")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub created_at: Option<NaiveDateTime>,

    #[serde(rename = "created_by")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub created_by: Option<String>,

    #[serde(rename = "updated_at")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub updated_at: Option<NaiveDateTime>,

    #[serde(rename = "updated_by")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub updated_by: Option<String>,

    #[serde(rename = "name")]
    pub name: String,

    #[serde(rename = "note")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub note: Option<String>,

    #[serde(rename = "country")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub country: Option<String>,

    #[serde(rename = "expiry_date")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub expiry_date: Option<String>,

    #[serde(rename = "identity_card_number")]
    pub identity_card_number: String,

    #[serde(rename = "identity_card_type")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub identity_card_type: Option<String>,

    #[serde(rename = "issue_date")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub issue_date: Option<String>,

    #[serde(rename = "name_on_card")]
    pub name_on_card: String,

    #[serde(rename = "state")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub state: Option<String>,
}
